src/engine.h

```diff
@@ -87,6 +96,9 @@ class EngineController {
   std::string network_path_;
   std::string backend_;
   std::string backend_options_;
+
+  optional<CurrentPosition> current_position_;
```

Comment:
What does current_position_ really contain, and how is it different from tree_.GetPositionHistory()? Could you add a comment?

Reply:
That's a very good question. It made me realize that a) this should really be called something like `next_search_position_`, especially since b) this really needs to be reset when doing a normal search, otherwise repeated `go` commands will just search the same position instead of playing a full game.

I'll push a commit to address your comments soon.

Reply:
Well, looks like I misremembered how UCI works and the implementation as is was actually fine. (repeated `go` commands should in fact just search the same position over and over.) So I kept the name as is and only added a comment to document that this position is the same as the position in the search tree, except during ponder search.